### PR TITLE
Fix free5GC tests and dedup SMF configrefs

### PR DESCRIPTION
**Pull request by tliron**
Also:

* Update README to install ref.nephio.org Config
* Upgrade to Go 1.21.6
* Remove old and unused CRD files
* Match AMF code style to that of the other NFs
* Small fixes to log and error messages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,8 +13,17 @@ Getting Started
 
 Prior to running the free5GC operator, Multus needs to be installed on cluster with the macvlan CNI.
 
+You will also need the Nephio API CRDs, from the [api repository](https://github.com/nephio-project/api):
+
+```sh
+git clone https://github.com/nephio-project/api
+kubectl apply -f api/config/crd/bases/ref.nephio.org_configs.yaml
```
Comment (arora-sagar) [inline README.md:20]: @tliron we can do this instead

```
kubectl apply -f https://raw.githubusercontent.com/nephio-project/api/main/config/crd/bases/ref.nephio.org_configs.yaml 

kubectl apply -f https://raw.githubusercontent.com/nephio-project/api/main/config/crd/bases/workload.nephio.org_nfdeployments.yaml

kubectl apply -f https://github.com/nephio-project/api/blob/main/config/crd/bases/workload.nephio.org_nfconfigs.yaml
```

File: test/upf2-n3.yaml
Comment (arora-sagar) [inline test/upf2-n3.yaml:1]: @tliron Don't we need multus definitions for testing the UPF?

File: test/smf1-config-ref.yaml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+  name: smf1-config-ref
+
 spec:
   config:
     apiVersion: workload.nephio.org/v1alpha1
     kind: NFDeployment
     metadata:
-      name: free5gc-upf-1
-      namespace: upf-1
+      namespace: smf-1
```
Comment (lapentad) [inline test/smf1-config-ref.yaml:13]: Hi in the old version it was "free5gc-upf-1" should we keep UPF here? I think smf config has reference of the upf?

File: README.md (hunk 1 above)
Comment (tliron) [reply README.md:20]: Ah... this is much better. Actually I should remove the CRD dupes from the free5gc repo! And also fix the Makefile to not install these dupes.

File: test/upf2-n3.yaml
Comment (tliron) [reply test/upf2-n3.yaml:1]: Hm, I didn't change this from the original. Actually I didn't test that UPF works, only that it stands up.

Comment (arora-sagar) [reply test/upf2-n3.yaml:1]: Then let's try it and then delete this part. I think for the block test we will need it.

File: controllers/nf/amf/resources.go [hunk 3]
```diff
@@ -132,9 +132,9 @@ func createDeployment(log logr.Logger, configMapVersion string, nfDeployment *ne
 	return deployment, nil
 }
 
-func createService(nfDeployment *nephiov1alpha1.NFDeployment) *apiv1.Service {
-	namespace := nfDeployment.Namespace
-	instanceName := nfDeployment.Name
+func createService(amdDeployment *nephiov1alpha1.NFDeployment) *apiv1.Service {
+	namespace := amdDeployment.Namespace
+	instanceName := amdDeployment.Name
```
Comment (electrocucaracha) [inline controllers/nf/amf/resources.go:137]:
```suggestion
func createService(amfDeployment *nephiov1alpha1.NFDeployment) *apiv1.Service {
	namespace := amfDeployment.Namespace
	instanceName := amfDeployment.Name
```

File: controllers/nf/amf/reconciler.go [hunk 4]
```diff
@@ -172,7 +172,7 @@ func (r *AMFDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			}
 		}
 	} else {
-		log.Error(err, fmt.Sprintf("Failed to create Deployment %s\n", err.Error()))
+		log.Error(err, fmt.Sprintf("Failed to create Deployment %s", err.Error()))
```
Comment (electrocucaracha) [inline controllers/nf/amf/reconciler.go:175]:
```suggestion
		log.Error(err, fmt.Sprintf("Failed to create AMF NFDeployment %s", err.Error()))
```

File: controllers/nf/amf/resources.go (hunk 3 above)
Comment (tliron) [reply controllers/nf/amf/resources.go:137]: Oops :)

File: controllers/nf/amf/reconciler.go (hunk 4 above)
Comment (tliron) [reply controllers/nf/amf/reconciler.go:175]: Actually, this log error is about failure to creating a K8s Deployment resource, I will keep the original.

**Pull request merged**